pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -84,28 +83,13 @@ func add(mgr manager.Manager, r *ControlPlaneReconciler) error {
 	}
 
 	// watch created resources for use in synchronizing ready status
-	if err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &v2.ServiceMeshControlPlane{},
-		},
-		ownedResourcePredicates); err != nil {
+	if err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, enqueueRequestForSMCP, ownedResourcePredicates); err != nil {
 		return err
 	}
-	if err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &v2.ServiceMeshControlPlane{},
-		},
-		ownedResourcePredicates); err != nil {
+	if err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}}, enqueueRequestForSMCP, ownedResourcePredicates); err != nil {
```

Comment:
nit: not sure why we have this even. maybe we should file an issue to remove it?

Reply:
Agreed.  This is only necessary if the control plane includes stateful sets.